operator/src/main/java/org/keycloak/operator/controllers/KeycloakIngress.java

```diff
@@ -79,6 +83,7 @@ private Ingress newIngress() {
                         .endService()
                     .endDefaultBackend()
                     .addNewRule()
+                        .withHost(keycloak.getSpec().getHostnameSpec().getHostname())
```

Comment:
Comment unrelated to this line. I believe we should not set this if `hostname == null`.